### ISSUE TITLE
Auto start service

### ISSUE THREAD

**Issue by PoD-Stas** [tananaev/traccar-client-ios#35]
Hi!
I have jailbreaked devices. and I need to automatically start service.
Don't u know how can I modify this app, or make some tweak for cydia to do that on my device? :)
Or maybe I can modify deployed app to do that?

**Comment by tananaev** [comment]
You can use `voip` background mode. Here is a quote from Apple documentation:
> The app provides Voice-over-IP services. Apps with this key are automatically launched after system boot so that the app can reestablish VoIP services. Apps with this key are also allowed to play background audio.

**Comment by tacman** [comment]
Is this the only way to do that?

**Comment by tananaev** [comment]
Yes, it is the only way.

**Comment by tacman** [comment]
How about significant location change?  That's also a valid background / auto-launch, right?

**Comment by tananaev** [comment]
Background location updates mode doesn't auto-start an app.

**Comment by tacman** [comment]
If the app is running in the background and the phone is rebooted, you have
to manually re-open the app and start the service?  I thought in the past
this ran continuously.   Am I misunderstanding something?

**Comment by tananaev** [comment]
Yes, you have to re-open the app. Only Traccar Client for Android has auto-start option.

**Comment by PoD-Stas** [comment]
So the only reason is Apple politics about privacy? Technically for personal use do I just have to enable "voip" background mode to the [main plist](https://developer.apple.com/library/ios/documentation/Performance/Conceptual/EnergyGuide-iOS/Art/xcode_project_capabilities_backgroundmodes_2x.png) to add?
And then launch
`self.trackingController = [[TCTrackingController alloc] init];
        [self.trackingController start];`

From for example `- (void)viewDidAppear`

https://developer.apple.com/library/ios/documentation/Performance/Conceptual/EnergyGuide-iOS/Art/xcode_project_capabilities_backgroundmodes_2x.png

**Comment by tananaev** [comment]
I guess you can do it for your own developer build, but it will never be approved for the App Store.

**Comment by PoD-Stas** [comment]
Okay thanks! Yes it's just for my personal use.

**Comment by jeanlk15** [comment]
can I ask if the auto start of traccar is still an issue on iphones ?
